contracts/cw-proposal-single/src/contract.rs

```diff
@@ -715,14 +727,16 @@ pub fn migrate(_deps: DepsMut, _env: Env, _msg: MigrateMsg) -> Result<Response,
 
 #[cfg_attr(not(feature = "library"), entry_point)]
 pub fn reply(deps: DepsMut, _env: Env, msg: Reply) -> Result<Response, ContractError> {
-    if msg.id % 2 == 0 {
+    if msg.id == FAILED_PROPOSAL_REPLY_ID {
+        Ok(Response::new())
+    } else if msg.id % 2 == 0 {
         // Proposal hook so we can just divide by two for index
-        let idx = msg.id / 2;
+        let idx = msg.id / 2 - 1;
```

Comment:
just curious, why did you have to change this?

Reply:
I had to reserve some reply id where we don't delete hook, decided it can be 0, is there any better way?

Reply:
Even / odd worked really well when there were only two options.

One thing we might try is some bitmasking. Could make the first couple bits in the ID an identifier for what kind of reply we're dealing with.